beer-garden/beer-garden
# #1306 - Protected roles

Closes #1306 

This PR adds the concept of "protected" roles and makes "superuser" one such role.  A protected role is one that is intended to be internally managed and is therefore not modified or deleted during any role imports or syncs.

## Test Instructions
* Start your garden.  The superuser role will be created / configured as necessary.
* Set your auth -> role_definition_file setting to point to `example_configs/roles.yaml`
* Restart your garden and note that the "superuser" role stays intact, despite it not being present in the `roles.yaml`.
* Add the "superuser" role to the roles.yaml.  For example:
  ```yaml
    - name: "superuser"
      permissions:
        - "garden:read"
    ```
* Restart your garden and note that the "superuser" role again has not changed.  If you look at the beer-garden logs you should see the message "Role sync request for protected role superuser will be ignored," informing you that "superuser" was not changed even though it was present in the `roles.yaml`.
